core/src/main/java/jenkins/views/PartialHeader.java

```diff
@@ -0,0 +1,49 @@
+package jenkins.views;
+
+import java.util.logging.Logger;
+
+import hudson.ExtensionList;
+import hudson.init.InitMilestone;
+import hudson.init.Initializer;
+import hudson.util.AdministrativeError;
+import jenkins.model.Jenkins;
+
+/**
+ * {@link Header} that relies on core resources (images, CSS, JS, etc.) to perform
+ * partial replacements.
+ * 
+ * Given this kind of header is not independent, compatibility should be managed by the
+ * specific {@link Header} compatibility header version value
+ * 
+ * @see Header
+ */
+public abstract class PartialHeader extends Header {
+
+    private static Logger LOGGER = Logger.getLogger(PartialHeader.class.getName());
+    
+    /**
+     * The current compatibility version of the Header API.
+     * 
+     * Increment this number when an incompatible change is made to the header (like the search form API).
+     */
+    private static final int compatibilityHeaderVersion = 1;
+    
+    @Override
+    public final boolean isCompatible() {
+        return compatibilityHeaderVersion == getSupportedHeaderVersion();
+    }
+    
+    /**
+     * @return the supported header version
+     */
+    public abstract int getSupportedHeaderVersion();
```

Comment:
Perhaps it's too restrictive to force "equal" for the compatibility. No strong opinion